src/main/java/at/hannibal2/skyhanni/features/inventory/bazaar/CraftMaterialsFromBazaar.kt

```diff
@@ -0,0 +1,166 @@
+    private var display = listOf<Renderable>()
+    private var neededMaterials = mapOf<NEUInternalName, Int>()
+    private var multiplier = 1
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        val correctInventoryName = pattern.matches(event.inventoryName)
+        val items = event.inventoryItems
+        val correctItem = items[23]?.name == "§aCrafting Table"
+
+        inRecipeInventory = correctInventoryName && correctItem && !purchasing
+        if (!inRecipeInventory) return
+
+        val recipeName = items[25]?.itemName ?: return
+        showRecipe(calculateMaterialsNeeded(items), recipeName)
+    }
+
+    private fun showRecipe(
+        recipeMaterials: MutableMap<NEUInternalName, Int>,
+        recipeName: String,
+    ) {
+        val neededMaterials = mutableMapOf<NEUInternalName, Int>()
+        display = buildList {
+            val totalPrice = calculateTotalPrice(recipeMaterials, 1)
+            add(Renderable.string("§7Craft $recipeName §7(§6${NumberUtil.format(totalPrice)}§7)"))
+            for ((material, amount) in recipeMaterials) {
+                var text = "§8${amount.addSeparators()}x " + material.itemName
+                if (material.isBazaarItem()) {
+                    neededMaterials[material] = amount
+                    text += " §6${NumberUtil.format(material.getPrice() * amount)}"
+                }
+                add(Renderable.string(text))
+            }
+            if (neededMaterials.isNotEmpty()) {
+                add(
+                    Renderable.clickAndHover(
+                        "§eGet from bazaar!",
+                        listOf("§eClick here to buy the items from bazaar!"),
+                        onClick = {
+                            getFromBazaar(neededMaterials)
+                        })
+                )
+            }
+        }
+    }
+
+    private fun calculateMaterialsNeeded(items: Map<Int, ItemStack>): MutableMap<NEUInternalName, Int> {
+        val recipeMaterials = mutableMapOf<NEUInternalName, Int>()
+        for (slot in matieralSlots) {
+            val item = items[slot] ?: continue
+            val internalName = item.getInternalNameOrNull() ?: continue
+            val size = item.stackSize
+            recipeMaterials.addOrPut(internalName, size)
+        }
+        return recipeMaterials
+    }
+
+    private fun getFromBazaar(neededMaterials: MutableMap<NEUInternalName, Int>) {
+        this.neededMaterials = neededMaterials
+        this.multiplier = 1
+        purchasing = true
+        update()
+    }
+
+    private fun update() {
+        display = buildList {
+            add(Renderable.string("§7Buy items from Bazaar:"))
+            for ((material, amount) in neededMaterials) {
+                val priceMultiplier = amount * multiplier
+                var text = "§8${priceMultiplier.addSeparators()}x " + material.itemName
+                if (material.isBazaarItem()) {
```

Comment:
the if is redundant since all materials are already filter for isBazaarItem in the assignment of the neededMaterials